src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,210 @@
+		syntaxJsonObject.add("description", convertArrayToJsonArray(eventInfo.getDescription()));
+		syntaxJsonObject.add("examples", convertArrayToJsonArray(eventInfo.getExamples()));
+		syntaxJsonObject.add("patterns", convertToJsonArray(eventInfo.patterns));
+		return syntaxJsonObject;
+	}
+
+
+	private <T extends StructureInfo<? extends Structure>> JsonArray generateStructureElementArray(Iterator<T> infos) {
+		JsonArray syntaxArray = new JsonArray();
+		infos.forEachRemaining(info -> {
+			if (info instanceof SkriptEventInfo) {
+				syntaxArray.add(generateEventElement((SkriptEventInfo<?>) info));
+			} else {
+				JsonObject structureElementJsonObject = generatedAnnotatedElement(info.getElementClass(), info.patterns);
+				if (structureElementJsonObject != null)
+					syntaxArray.add(structureElementJsonObject);
+			}
+		});
+		return syntaxArray;
+	}
+
+	private <T extends SyntaxElementInfo<? extends SyntaxElement>> JsonArray generateSyntaxElementArray(Iterator<T> infos) {
+		JsonArray syntaxArray = new JsonArray();
+		infos.forEachRemaining(info -> {
+			JsonObject syntaxJsonObject = generatedAnnotatedElement(info.getElementClass(), info.patterns);
+			if (syntaxJsonObject != null)
+				syntaxArray.add(syntaxJsonObject);
+		});
+		return syntaxArray;
+	}
+
+	private @Nullable JsonObject generateClassInfoElement(ClassInfo<?> classInfo) {
+		if (!classInfo.hasDocs())
+			return null;
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", idProvider.getId(classInfo));
+		syntaxJsonObject.addProperty("name", classInfo.getDocName());
+		syntaxJsonObject.addProperty("since", classInfo.getSince());
+		syntaxJsonObject.add("description", convertToJsonArray(classInfo.getDescription()));
+		syntaxJsonObject.add("examples", convertToJsonArray(classInfo.getExamples()));
+		syntaxJsonObject.add("patterns", convertToJsonArray(classInfo.getUsage()));
+		return syntaxJsonObject;
+	}
+
+	private JsonArray generateClassInfoArray(Iterator<ClassInfo<?>> classInfos) {
+		JsonArray syntaxArray = new JsonArray();
+		classInfos.forEachRemaining(classInfo -> {
+			JsonObject classInfoElement = generateClassInfoElement(classInfo);
+			if (classInfoElement != null)
+				syntaxArray.add(classInfoElement);
+		});
+		return syntaxArray;
+	}
+
+	private JsonObject generateFunctionElement(JavaFunction<?> function) {
+		JsonObject functionJsonObject = new JsonObject();
+		functionJsonObject.addProperty("id", idProvider.getId(function));
+		functionJsonObject.addProperty("name", function.getName());
+		functionJsonObject.addProperty("since", function.getSince());
+		functionJsonObject.add("description", convertToJsonArray(function.getDescription()));
+		functionJsonObject.add("examples", convertToJsonArray(function.getExamples()));
+
+		ClassInfo<?> returnType = function.getReturnType();
+		if (returnType != null) {
+			functionJsonObject.addProperty("return-type", returnType.getDocName());
```

Comment:
Java types return-type is empty in json
![image](https://github.com/SkriptLang/Skript/assets/20037329/0786370d-cd68-4742-b2da-eee3866354dd)

See how it's handled in HTMLGen
https://github.com/SkriptLang/Skript/blob/1a009cdb1419292046de0e95429bcb40eaaaab15/src/main/java/ch/njol/skript/doc/HTMLGenerator.java#L919